Changelog.md

```diff
@@ -1,26 +1,39 @@
 # Change Log
 ## bpaf [0.6.0] - unreleased
-- removed `positional_os()` and `argument_os()`
-- removed `from_str`
-- internal refactors
-- documentation refactor: examples with input/output
-- `any` - a positional way to capture anything
-- `adjacent` restriction
-- `catch` inside `ParseOption`, `ParseMany` and `ParseSome`
-- usage line refactor, cosmetic mostly
-- include \n in --version, cosmetic
-
-Migration guide:
-1. replace `positional_os("FOO")` with `positional::<Type>("FOO")` or use `positional` in derive API
-2. replace `argument_os("FOO")` with `argument::<Type>("FOO")` or use `argument` in derive API
-3. replace any uses of `from_str::<T>` with turbofish on the nearest `positional` or
-   `argument`. You can use `map` to apply `FromStr` parser if `String` gets created inside your parsers
-
-You shouldn't be using those names directly, but there are some structure renames:
-1. `Positional` -> `ParsePositional`
-2. `BuildArgument` -> ParseArgument`
-3. `Command` -> `ParseCommand`
-
+# What's new in 0.6.0
+- `adjacent` restriction to parse things in a tighter context
+- `catch` for `many`, `some` and `optional` to handle parse errors
+- `any` positional like, to consume pretty much anything from a command line
+- improved documentation with more detailed examples
+- cosmetic improvements
+- a sneaky reminder to use `to_options` on `Parser` before trying to run it
+- a way to make boxed parsers with single item `construct!` macro for making dynamic parsers
+- a horrible way to reduce Yoda-talk coding by placing primitive definitions inside the `construct!` macro
+
+With new additions you should be able to parse pretty much anything and then some more :)
+
```

Comment:
Shouldn't `cargo_helper` be in batteries?

Also, the derive example seems unclear to me.

> If you pass a cargo command name as a parameter to options annotation bpaf_derive would generate cargo_helper.

but I see nothing in the example code that looks like I'm passing a cargo command name in

Reply:
> Shouldn't cargo_helper be in batteries?

Ideally - yes, but that's optional and `bpaf_derive` uses it unconditionally. Will see what I can do about it. And about the example too

Reply:
imo this is too specialized to build directly into the derive especially without a direct reference to the feature.

Reply:
Maybe. I guess I'll define it in the main library, hide and reexport from batteries. That's all the helper FWIW, with `catch` it got much cleaner: 
1. Want to parse a word
2. Word must match a given string literal
3. It's okay if it's missing
4. It's also okay when it's not matching expectations
5. And don't show anything to the user
6. Parse this word and then everything else as a tuple, return that second item.
```rust
pub fn cargo_helper<P, T>(cmd: &'static str, parser: P) -> impl Parser<T>
where
    T: 'static,
    P: Parser<T>,
{
    let skip = positional::<String>("cmd")
        .guard(move |s| s == cmd, "")
        .optional()
        .catch()
        .hide();
    construct!(skip, parser).map(|x| x.1)
}
```